c7n/query.py

```diff
@@ -475,11 +475,22 @@ def __init__(self, ctx, data):
 
     @property
     def source_type(self):
-        return self.data.get('source', 'describe')
+        value = self.data.get('source', 'describe')
+        if isinstance(value, str):
+            return value
+        # Here we assume that the data has been validated, so the other
+        # alternative is that the source is a dict with the name attribute
+        # being a valid source type.
+        return value["name"]
+
+    @property
+    def source_config(self):
```

Comment:
It does seem useful to have a separate property for the configuration, though I see some potential confusion between `source_config` and the config source. Definitely a nit, and possibly unavoidable.

Still it's possible that something like `source_settings` or `source_attributes` could also work? :shrug: